keps/sig-node/4753-expose-ownerref-downard-api/README.md

```diff
@@ -0,0 +1,616 @@
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+  ownerReferences:
+  - apiVersion: apps/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: DaemonSet
+    name: an-owned-pod-1722852739
+    uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+  - apiVersion: apps/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: Some-CRD
+    name: an-owned-pod-1722852739
+    uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+spec:
+  containers:
+    - name: client-container
+      image: busybox
+      command: ["sh", "-c"]
+      args:
+      - while true; do
+          if [[ -e /etc/podinfo/ownerReferences ]]; then
+            echo -en '\n\n'; cat /etc/podinfo/ownerReferences; fi;
+          sleep 5;
+        done;
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc/podinfo
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "ownerReferences"
+            fieldRef:
+              fieldPath: metadata.ownerReferences
+```
+
+Logs from the pod will show the ownerReferences of the pod:
+
+```bash
+$ kubectl logs kubernetes-downwardapi-volume-example
+- apiVersion: apps/v1
+  blockOwnerDeletion: true
+  controller: true
+  kind: DaemonSet
+  name: an-owned-pod-1722852739
+  uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+- apiVersion: apps/v1
+  blockOwnerDeletion: true
+  controller: true
+  kind: Some-CRD
+  name: an-owned-pod-1722852739
+  uid: 6be1683f-da9c-4f68-9440-82376231cfa6
+```
+
+If `ownerReferences` is empty or not set, the file will be empty.
+
+Is out of the scopt of this KEP to add the `ownerReferences` to the pod's environment variables.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+Unit and e2e testing will be added consistent with other resources in downward API.
+
+##### Integration tests
+
+There are no integration tests, only e2e tests.
+
+##### e2e tests
+
+e2e testing will consist of the following:
+- Create a Deployment and or DaemonSet
+- Verify if the pod created by the Deployment/DaemonSet has the ownerReferences of the Deployment/DaemonSet via ENV VAR
```

Comment:
is this correct? you said before env variables are out of scope

> Is out of the scopt of this KEP to add the `ownerReferences` to the pod's environment variables.

Reply:
my bad fixed.

Reply:
fixed